src/App.tsx

```diff
@@ -1,13 +1,49 @@
 /* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
-import { USER_ID } from './api/todos';
+import React, { useEffect, useState } from 'react';
+// import { UserWarning } from './UserWarning';
+import { getTodos } from './api/todos';
+import { Todo } from './types/Todo';
+import { ToodoList } from './components/TodoList';
+import { FilterOptions } from './utils/FilterOptions';
+import cn from 'classnames';
+
+function getFilteredTodos(todos: Todo[], filterOption: FilterOptions): Todo[] {
+  return todos.filter(todo => {
+    switch (filterOption) {
+      case FilterOptions.Active:
+        return !todo.completed;
+
+      case FilterOptions.Completed:
+        return todo.completed;
+
+      default:
+        return true;
+    }
+  });
+}
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todoos, setTodos] = useState<Todo[]>([]);
+  const [error, setError] = useState('');
+  const [filterOption, setFilterOption] = useState<FilterOptions>(
+    FilterOptions.All,
+  );
+
+  useEffect(() => {
+    getTodos()
+      .then(setTodos)
+      .catch(() => setError('Unable to load todos'))
+      .finally(() =>
+        setTimeout(() => {
+          setError('');
+        }, 3000),
+      );
+  }, []);
+
+  const numberOfActiveTodos = todoos.filter(todo => !todo.completed).length;
```

Comment:
This is an example of an operation that could be wrapped in useMemo, but it's all up to you

Reply:
yep, u need `useMemo` in this case.
With `useMemo`, u will avoid re-creating the variable at each rendering